File: Eazy_Store/src/main/java/com/eazybytes/eazystore/service/OrderService.java
package com.eazybytes.eazystore.service;

import java.util.List;

import com.eazybytes.eazystore.dto.OrderResponse;
import com.eazybytes.eazystore.dto.OrderStatusRequest;
import com.eazybytes.eazystore.dto.OrderStatusRequest;

public interface OrderService {

    OrderResponse placeOrder();

    List<OrderResponse> getOrders();

    OrderResponse getOrderById(Long orderId);
    OrderResponse updateOrderStatus(Long orderId,
            OrderStatusRequest request);
    List<OrderResponse> getAllOrders();
}

File: frontend/src/services/orderService.js
import apiClient from "../api/apiClient";

const placeOrder = () => {
  return apiClient.post("/orders");
};

const getOrders = () => {
  return apiClient.get("/orders");
};

const getAllOrders = () => {
  return apiClient.get("/orders/admin");
};

const getOrderById = (orderId) => {
  return apiClient.get(`/orders/${orderId}`);
};

// NEW
const updateOrderStatus = (orderId, status) => {
  return apiClient.put(`/orders/${orderId}/status`, {
    status,
  });
};

export default {
  placeOrder,
  getOrders,
  getAllOrders,
  getOrderById,
  updateOrderStatus,
};
